### PR TITLE
takeLast: fix overload signature and add related benchmark

### PR DESCRIPTION
**Pull request by gcanti**
/cc @mfirry

### CODE REVIEW

File: src/takeLast.ts [hunk 1]
```diff
@@ -1,13 +1,12 @@
 import { takeRight } from 'fp-ts/lib/Array';
 
-export function takeLast<A>(a: number): (as: Array<A>) => Array<A>;
-export function takeLast<A>(a: number, as: Array<A>): Array<A>;
-export function takeLast<A>(a: number, as?: Array<A>): any {
+export function takeLast(i: number): <A>(as: Array<A>) => Array<A>;
```
Comment (giogonzo) [inline src/takeLast.ts]: 👍

File: dtslint/ts3.5/tslint.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "extends": "dtslint/dtslint.json",
+  "rules": {}
+}
```
Comment (giogonzo) [inline dtslint/ts3.5/tslint.json]: I tried to run `npm test` locally without this file and everything still works fine. If not strictly needed, I'd prefer to remove this

Comment (giogonzo) [reply dtslint/ts3.5/tslint.json]: discussed offline.
Never set up properly tslint in my VSCode. This file helps in getting an inline preview of failing `$Expect` assertions

**Comment by giogonzo** [comment]
thank you @gcanti

File: dtslint/ts3.5/tslint.json (hunk 2 above)
Comment (gcanti) [reply dtslint/ts3.5/tslint.json]: @giogonzo looks like this file is required by the [ms-vscode.vscode-typescript-tslint-plugin](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-tslint-plugin) extension in order to show the failing type-level tests in VSCode

btw I noticed that there's no configured tslint in this project, I'll send another PR for that

**Pull request merged**